moshi/src/main/java/com/squareup/moshi/JsonReader.java

```diff
@@ -445,6 +445,16 @@ public final Object readJsonValue() throws IOException {
   }
 
   /**
+   * Consumes the value of the next token and returns a JsonReader to defer reading the value.
+   *
+   * @throws JsonDataException if the next token is not a literal value, if a JSON object has a
+   * duplicate key.
+   */
+  public final JsonReader bufferValue() throws IOException {
+    return new JsonValueReader(readJsonValue());
```

Comment:
This might be better exposed simply as JsonReader.of(Object). Awkwardly, any API here will render JsonAdapter.fromValue(Object) redundant.